assume/common/grid_utils.py

```diff
@@ -235,22 +245,15 @@ def read_pypsa_grid(
     """
 
     def add_buses(network: pypsa.Network, buses: pd.DataFrame) -> None:
-        network.madd(
-            "Bus",
-            names=buses.index,
-            **buses,
-        )
+        network.import_components_from_dataframe(buses, "Bus")
 
     def add_lines(network: pypsa.Network, lines: pd.DataFrame) -> None:
-        network.madd(
-            "Line",
-            names=lines.index,
-            **lines,
-        )
+        network.import_components_from_dataframe(lines, "Line")
 
     # setup the network
     add_buses(network, grid_dict["buses"])
     add_lines(network, grid_dict["lines"])
+    add_loads(network, grid_dict["loads"])
```

Comment:
this shouldn't be here since we define different loads for different types of network. we have loads for redispatch and for nodal and these use different functions